Source/Client.swift

```diff
@@ -126,6 +128,9 @@ extension Client {
                 }
 
                 return response
+            } else if data.count == 0 {
+                //no data received, connection got broken
+                throw ClientError.brokenConnection
```

Comment:
And yeah the name should be different, after you investigate please let me know which one you want there (or push it directly).